File: ts-sdk/src/sdk/models/shared/metadataquery.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";


export enum MetadataQueryOrderByDirectionEnum {
    AscUpper = "ASC",
    DescUpper = "DESC",
    AscLower = "asc",
    DescLower = "desc"
}


// MetadataQueryOrderBy
/** 
 * An object representing one of the metadata template fields to sort the
 * metadata query results by.
**/
export class MetadataQueryOrderBy extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=direction" })
  direction?: MetadataQueryOrderByDirectionEnum;

  @SpeakeasyMetadata({ data: "json, name=field_key" })
  fieldKey?: string;
}


// MetadataQuery
/** 
 * Create a search using SQL-like syntax to return items that match specific
 * metadata.
**/
export class MetadataQuery extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=ancestor_folder_id" })
  ancestorFolderId: string;

  @SpeakeasyMetadata({ data: "json, name=fields" })
  fields?: string[];

  @SpeakeasyMetadata({ data: "json, name=from" })
  from: string;

  @SpeakeasyMetadata({ data: "json, name=limit" })
  limit?: number;

  @SpeakeasyMetadata({ data: "json, name=marker" })
  marker?: string;

  @SpeakeasyMetadata({ data: "json, name=order_by", elemType: MetadataQueryOrderBy })
  orderBy?: MetadataQueryOrderBy[];

  @SpeakeasyMetadata({ data: "json, name=query" })
  query?: string;

  @SpeakeasyMetadata({ data: "json, name=query_params" })
  queryParams?: Record<string, string>;
}

File: go-sdk/pkg/models/shared/metadataquery.go
package shared

type MetadataQueryOrderByDirectionEnum string

const (
	MetadataQueryOrderByDirectionEnumAscUpper  MetadataQueryOrderByDirectionEnum = "ASC"
	MetadataQueryOrderByDirectionEnumDescUpper MetadataQueryOrderByDirectionEnum = "DESC"
	MetadataQueryOrderByDirectionEnumAscLower  MetadataQueryOrderByDirectionEnum = "asc"
	MetadataQueryOrderByDirectionEnumDescLower MetadataQueryOrderByDirectionEnum = "desc"
)

// MetadataQueryOrderBy
// An object representing one of the metadata template fields to sort the
// metadata query results by.
type MetadataQueryOrderBy struct {
	Direction *MetadataQueryOrderByDirectionEnum `json:"direction,omitempty"`
	FieldKey  *string                            `json:"field_key,omitempty"`
}

// MetadataQuery
// Create a search using SQL-like syntax to return items that match specific
// metadata.
type MetadataQuery struct {
	AncestorFolderID string                 `json:"ancestor_folder_id"`
	Fields           []string               `json:"fields,omitempty"`
	From             string                 `json:"from"`
	Limit            *int64                 `json:"limit,omitempty"`
	Marker           *string                `json:"marker,omitempty"`
	OrderBy          []MetadataQueryOrderBy `json:"order_by,omitempty"`
	Query            *string                `json:"query,omitempty"`
	QueryParams      map[string]string      `json:"query_params,omitempty"`
}
